File: compiler/bench/simd2.fs
\ expected: 666566670000
\ Parallel multiply-add - vectorizable fused multiply-add

create a 80000 allot
create b 80000 allot

: init ( -- )
  10000 0 do
    i i 8 * a + !
    i 2 * i 8 * b + !
  loop ;

: dotprod ( -- n )
  0 10000 0 do
    i 8 * a + @ i 8 * b + @ * +
  loop ;

: main init dotprod . cr ;

File: compiler/bench/simd2.c
// expected: 666566670000
// Parallel multiply-add - vectorizable fused multiply-add
#include <stdio.h>
#define SIZE 10000
long a[SIZE], b[SIZE];

int main() {
    for (int i = 0; i < SIZE; i++) { a[i] = i; b[i] = i * 2; }
    long sum = 0;
    for (int i = 0; i < SIZE; i++) sum += a[i] * b[i];
    printf("%ld\n", sum);
    return 0;
}
